lib/openai/real_time.rb

```diff
@@ -0,0 +1,31 @@
+require "faye/websocket"
+require "eventmachine"
+
+module OpenAI
+  class RealTime
+    include HTTPHeaders
+
+    def initialize(client:)
+      @client = client
+      @websocket = nil
+      @on_message = nil
+    end
+
+    def on_message(&block)
+      @on_message = block
+    end
+
+    def connect(model: "gpt-4o-realtime-preview-2024-10-01")
+      uri = "#{File.join(@client.websocket_uri_base, @client.api_version, 'realtime')}?model=#{model}"
+
+      EM.run do
+        @websocket = Faye::WebSocket::Client.new(uri, nil, headers: openai_realtime_headers)
+        @websocket.on :message, @on_message
+      end
```

Comment:
Perhaps we should replace [Eventmachine](https://github.com/eventmachine/eventmachine) since it last release was 6 yeas ago. I think we could use [async](https://github.com/socketry/async). I have tried and it works fine with something like this:
```
    Async do
      endpoint = Async::HTTP::Endpoint.parse(uri, alpn_protocols: Async::HTTP::Protocol::HTTP11.names)
      Async::WebSocket::Client.connect(endpoint, headers: @headers) do |connection|
        @websocket = connection

        while (message = connection.read)
          @on_message
        end
      end
    end
```